content/learn/intermediate/outputs-2.md

```diff
@@ -0,0 +1,249 @@
++++
+title = "Outputs 2"
+weight = 310
++++
+
+Until now, we have just used the [default response behavior](/learn/beginner/outputs-1) to make our custom commands
+respond with some text. This makes sense for quick mockups or relatively simple commands. However, this may be
+inconvenient or Not What You Want in some cases. In this chapter, we will explore how to send messages to different
+channels, edit already sent messages, and send messages with embeds, vastly expanding your toolbox for creating complex
+custom command systems.
+
+## Sending Messages
+
+We provide several `sendMessage*` functions that all do the same thing: send a message to a channel. The difference
+between them is how they handle special mentions like `@everyone` and `@here`, and whether to return the ID of the
+message after sending it. We will cover these functions in detail in the following sections, and when to best use them.
+
+Let's get started with the simplest of them all, `sendMessage`. Its syntax is the following:
+
+```go
+{{ sendMessage channel_id message_to_be_sent }}
+```
+
+The `channel_id` is the ID of the channel in which the message is to be sent. If you want to send the message in the
+same channel as the one in which the custom command was triggered, simply set `channel_id` to `nil`.
+
+Intuitively, `message_to_be_sent` denotes the output that is to be sent as a message. For now, we will just say it is a
+string with the content you want to send. We will cover sending embeds in a later section on this page.
+
+### Special Mentions
+
+By default, the bot will escape special mentions like `@everyone`, `@here`, and role mentions. If you want to send a
+message with these mentions, you'll need to tell the bot to not escape them. You do this by using the
+`sendMessageNoEscape` function instead of `sendMessage`.
+
+```go
+{{ sendMessageNoEscape channel_id message_to_be_sent }}
+```
+
+### Returning the Message ID
+
+If you want to store the ID of the message you just sent, for example to later edit it, you use the `sendMessageRetID`
+function. Assign the result of this function to a variable, otherwise it will be rendered to the output and become
+unusable for further usage in your custom command.
+
+```go
+{{ $messageID := sendMessageRetID channel_id message_to_be_sent }}
+```
+
+Naturally, we provide a combination of the `sendMessageRetID` function with the `sendMessageNoEscape` function if you
+want to send a message with special mentions and store the ID of the message.
```

Comment:
we provide a variant of the `sendMessageRetID` function that does not escape mentions: